schema/token.graphql

```diff
@@ -0,0 +1,384 @@
+enum TokenStatus {
+  SALE
+  MARKET
+  IDLE
+}
+
+type TokenAvatarObject @variant {
+  "The avatar data object"
+  avatarObject: StorageDataObject!
+}
+
+type TokenAvatarUri @variant {
+  "The avatar URL"
+  avatarUri: String!
+}
+
+union TokenAvatar = TokenAvatarObject | TokenAvatarUri
+
+type Token @entity {
+  "runtime token identifier"
+  id: ID!
+
+  "status sale / market / idle"
+  status: TokenStatus!
+
+  "avatar object (profile picture)"
+  avatar: TokenAvatar
+
+  "total supply"
+  totalSupply: BigInt!
+
+  "Whether a sale is ongoing"
+  sale: [Sale!] @derivedFrom(field: "token")
+
+  "symbol for the token uniqueness guaranteed by runtime"
+  symbol: String! @index @unique
+
+  "access status invite only vs anyone"
+  isInviteOnly: Boolean!
+
+  "creator annual revenue percent (minted)"
+  annualCreatorReward: BigInt!
+
+  "revenue share ratio between creator and holder"
+  revenueShareRatioPercent: Int!
+
+  "current revenue share id if any"
+  revenueShare: [RevenueShare!] @derivedFrom(field: "token")
+
+  "amm curve setup for token"
+  ammCurve: [AmmCurve!] @derivedFrom(field: "token")
+
+  "block in which this token was created"
+  createdAt: DateTime!
+
+  "channel from which the token is issued uniqueness guaranteed by runtime"
+  channel: TokenChannel @derivedFrom(field: "token")
+
+  "video for the token presentation page"
+  trailerVideo: Video
+
+  "about information displayed under the presentation video"
+  description: String
+
+  "list of benefits for the token"
+  benefits: [Benefit!] @derivedFrom(field: "token")
+
+  "note from creator to member interested in joining the whitelist"
+  whitelistApplicantNote: String
+
+  "link for creator to member interested in joining the whitelist"
+  whitelistApplicantLink: String
+
+  "holders having some amount of this token"
+  accounts: [TokenAccount!] @derivedFrom(field: "token")
+
+  "number of accounts to avoid aggregate COUNT"
+  accountsNum: Int!
+
+  "number of completed Amm Sales"
+  ammNonce: Int!
+
+  "number of revenue shares issued"
+  revenueShareNonce: Int!
+
+  "whether it has been deissued or not"
+  deissued: Boolean!
+}
+
+type TokenChannel @entity @index(fields: ["token", "channel"], unique: true) {
```

Comment:
I had to make the TokenChannel entity to have a optional to optional relationship.
The token.channelId is optional because I cannot infer channel information on the TokenIssued event
I need the CreatorTokenIssued event from the channel pallet in order to set the channelId on token.